synapse/events/__init__.py

```diff
@@ -111,6 +111,10 @@ def __init__(self, internal_metadata_dict: JsonDict):
         # in the DAG)
         self.outlier = False
 
+        # Whether this event has a valid redaction event pointing at it (i.e.
+        # whether it should be redacted before giving to clients).
+        self.redacted_by: Optional[str] = None
```

Comment:
From https://github.com/matrix-org/synapse/pull/10119#discussion_r681126225:

> I'm super dubious about stuffing more things in here that don't form part of the `internal_metadata` column in the database.
> 
> Should we start a new attrs class for things like this, `stream_ordering` and `outlier`? (it also looks like `before`, `after`, `order` might fall into the same category).

> also: it seems confusing to have both a `redacted_by` and a `redacted` property.

Reply:
the Event class, I think.